src/transformers/processing_utils.py

```diff
@@ -409,6 +607,111 @@ def from_args_and_dict(cls, args, processor_dict: Dict[str, Any], **kwargs):
+                ```
+            4) kwargs passed during instantiation of a modality processor have fourth priority.
+                ```python
+                tokenizer = tokenizer_class(..., {"padding": "max_length"})
+                image_processor = image_processor_class(...)
+                processor(tokenizer, image_processor) # will pass max_length unless overriden by kwargs at call
+                ```
+            5) defaults kwargs specified at processor level have lowest priority.
+
+        Args:
+            ModelProcessorKwargs (`ProcessingKwargs`):
+                Typed dictionary of kwargs specifically required by the model passed.
+            text_kwargs (`TextKwargs`, *optional*):
+                Typed dictionary of kwargs inputs applied to the text modality processor, i.e. the tokenizer.
+            images_kwargs (`ImagesKwargs`, *optional*):
+                Typed dictionary of kwargs inputs applied to the images modality processor.
+            videos_kwargs (`VideosKwargs`, *optional*):
+                Typed dictionary of kwargs inputs applied to the videos modality processor.
+            audio_kwargs (`AudioKwargs`, *optional*):
+                Typed dictionary of kwargs inputs applied to the audio modality processor.
+            tokenizer_init_kwargs (`Dict`, *optional*):
+                Dictionary of kwargs the tokenizer was instantiated with and need to take precedence over other kwargs.
+
+        Returns:
+            output_kwargs (`Dict`):
+                Dictionary of per-modality kwargs to be passed to each modality-specific processor.
+
+        """
+
+        # Initialize dictionaries
+        output_kwargs = {
+            "text_kwargs": text_kwargs or {},
+            "images_kwargs": images_kwargs or {},
+            "audio_kwargs": audio_kwargs or {},
+            "videos_kwargs": videos_kwargs or {},
+            "common_kwargs": common_kwargs or {},
+        }
+
+        default_kwargs = {
+            "text_kwargs": {},
+            "images_kwargs": {},
+            "audio_kwargs": {},
+            "videos_kwargs": {},
+            "common_kwargs": {},
+        }
+
+        # get defaults from set model processor kwargs if they exist
+        for modality in default_kwargs:
+            default_kwargs[modality] = ModelProcessorKwargs._defaults.get(modality, {}).copy()
+        # then override with tokenizer-level arguments passed
+        if tokenizer_init_kwargs:
+            default_kwargs["text_kwargs"].update(
+                {k: v for k, v in tokenizer_init_kwargs.items() if k in ModelProcessorKwargs.text_kwargs}
+            )
+
+        # then get passed per-modality dictionaries if they exist
+        for modality in output_kwargs:
+            output_kwargs[modality] = {
+                **default_kwargs[modality],
+                **output_kwargs[modality],
+                **kwargs.pop(modality, {}),
```

Comment:
If we have `modality` in the signature should we pop `modality` form kwargs?
I mean `image_kwargs` argument is defined in the signature, so it will not appear in kwargs, am I missing something?

Reply:
It can pop in kwargs, if you do:
```python
common_kwargs = {"return_tensors": "pt"}
images_kwargs = {"crop_size": {"height": 214, "width": 214}}
text_kwargs = {"padding": "max_length", "max_length": 76}

all_kwargs = {"images_kwargs": images_kwargs, "text_kwargs": text_kwargs, "common_kwargs": common_kwargs}

inputs = processor(text=input_str, images=image_input, **all_kwargs, max_length=98)
```
Then, the value `max_length` will indeed appear in `kwargs` despite existing in `text_kwargs`, this is just after the `__call__`:

```bash
(Pdb) kwargs
{'max_length': 98}
(Pdb) text_kwargs
{'padding': 'max_length', 'max_length': 76}
```

However I think we don't need to `pop` it here since anyway we do it right after, the following `pop` should be enough. Hope that clarifies it!

Reply:
ah, you're right! missed this when porting from a per-processor to a generic util, will update.